src/dataviews/histogram-dataview-model.js

```diff
@@ -165,15 +191,22 @@ module.exports = DataviewModelBase.extend({
 
   _onColumnChanged: function () {
     this._originalData.set('column_type', this.get('column_type'));
-    this.set('aggregation', undefined, { silent: true });
 
-    this._resetFilter();
+    this.set({
+      aggregation: undefined,
+      offset: undefined
+    }, { silent: true });
+
+    this._reloadVisAndForceFetch();
+  },
+
+  _onOffsetChanged: function () {
     this._reloadVisAndForceFetch();
   },
 
   _calculateTotalAmount: function (buckets) {
     return _.reduce(buckets, function (memo, bucket) {
-      return memo + bucket.freq;
+      return bucket && memo + bucket.freq;
```

Comment:
In case there is a null bucket in the middle of the `buckets` array the result of this reduce is wrong because it'll return null as the whole memo, reseting the result.

See this JSBin with the example https://jsbin.com/godimot/1/edit?html,js,console